src/nvim/buffer.c

```diff
@@ -5327,10 +5328,10 @@ int bufhl_add_hl(buf_T *buf,
                  int hl_id,
                  linenr_T lnum,
                  colnr_T col_start,
-                 colnr_T col_end) {
-  static int next_src_id = 1;
+                 colnr_T col_end)
+{
   if (src_id == 0) {
-    src_id = next_src_id++;
+    src_id = (int)nvim_create_namespace((String)STRING_INIT);
```

Comment:
It feels slightly irritating to me that this is still called `src_id`, which does not transport the relation to the namespaces. The important point to me is that if you have a namespace, you can use it here without having to fear any collisions. Should that be renamed or at least be reflected in the docs?

Reply:
The docs are updated (or rather they _will_ be updated the next time gen_api_docs is run, maybe we could teach @marvim to do that when needed or at some regular time interval). But you are right I should have changed them here to, I will do it in the next relevant PR...

Reply:
How about doc/api.txt, is that also autogenerated?

Reply:
yes, by the `gen_api_docs.py` script I mentioned.

Reply:
Ah nice :+1: